NEWS.md

```diff
@@ -563,6 +563,11 @@ Deprecated or removed
     and/or shape depending on `opts...`. For an algebraic multiplicative identity,
     consider `one(A)` ([#24656]).
 
+  * `ones(dims...)` `and ones(T, dims)` have been deprecated. The replacements are
+    `fill(1.0, dims...)` and `fill(one(T), dims...)`, but other common patterns, such as
```

Comment:
Perhaps mention using literals also for other common types, e.g. `fill(1, dims...)` for `Int`s, `fill(1f0, dims...)` for `Float32`, e.g. `fill(1+0im, dims...)` for complex types, `fill(0x1, dims...)` for `UInt8`s, e.g. `fill(1m, dims...)` for unitful types, et cetera? And perhaps `fill(T(1), dims...)` as another alternative alongside `fill(one(T), dims...)`?